python/ray/rllib/es/es.py

```diff
@@ -192,154 +168,154 @@ def _init(self):
-        tlogger.record_tabular("EpLenMean", lengths_n2.mean())
+        tlogger.record_tabular("EvalEpRewMean",
+                               np.nan if len(all_eval_returns) == 0
+                               else np.mean(all_eval_returns))
+        tlogger.record_tabular("EvalEpRewStd",
+                               np.nan if len(all_eval_returns) == 0
+                               else np.std(all_eval_returns))
+        tlogger.record_tabular("EvalEpLenMean",
+                               np.nan if len(all_eval_lengths) == 0
+                               else np.mean(all_eval_lengths))
+
+        tlogger.record_tabular("EpRewMean", noisy_returns.mean())
+        tlogger.record_tabular("EpRewStd", noisy_returns.std())
+        tlogger.record_tabular("EpLenMean", noisy_lengths.mean())
 
         tlogger.record_tabular(
-            "Norm", float(np.square(self.policy.get_trainable_flat()).sum()))
+            "Norm", float(np.square(self.policy.get_weights()).sum()))
         tlogger.record_tabular("GradNorm", float(np.square(g).sum()))
         tlogger.record_tabular("UpdateRatio", float(update_ratio))
 
-        tlogger.record_tabular("EpisodesThisIter", lengths_n2.size)
+        tlogger.record_tabular("EpisodesThisIter", noisy_lengths.size)
         tlogger.record_tabular("EpisodesSoFar", self.episodes_so_far)
-        tlogger.record_tabular("TimestepsThisIter", lengths_n2.sum())
+        tlogger.record_tabular("TimestepsThisIter", noisy_lengths.sum())
         tlogger.record_tabular("TimestepsSoFar", self.timesteps_so_far)
 
-        tlogger.record_tabular("ObCount", ob_count_this_batch)
-
         tlogger.record_tabular("TimeElapsedThisIter", step_tend - step_tstart)
         tlogger.record_tabular("TimeElapsed", step_tend - self.tstart)
         tlogger.dump_tabular()
 
         info = {
-            "weights_norm": np.square(self.policy.get_trainable_flat()).sum(),
+            "weights_norm": np.square(self.policy.get_weights()).sum(),
             "grad_norm": np.square(g).sum(),
             "update_ratio": update_ratio,
-            "episodes_this_iter": lengths_n2.size,
+            "episodes_this_iter": noisy_lengths.size,
             "episodes_so_far": self.episodes_so_far,
-            "timesteps_this_iter": lengths_n2.sum(),
+            "timesteps_this_iter": noisy_lengths.sum(),
             "timesteps_so_far": self.timesteps_so_far,
-            "ob_count": ob_count_this_batch,
             "time_elapsed_this_iter": step_tend - step_tstart,
             "time_elapsed": step_tend - self.tstart
         }
 
         result = TrainingResult(
-            episode_reward_mean=returns_n2.mean(),
-            episode_len_mean=lengths_n2.mean(),
-            timesteps_this_iter=lengths_n2.sum(),
+            episode_reward_mean=np.mean(all_eval_returns),
+            episode_len_mean=np.mean(all_eval_lengths),
+            timesteps_this_iter=noisy_lengths.sum(),
```

Comment:
a few redundant calculations, would be nice to get rid of. Reasonably, one could include all above information in this TrainingResult, cut the `tlogger` code and paste it into the `Agent`. (Perhaps out of scope for this PR if you don't want to do it)

Reply:
Let's clean this part up in a subsequent PR.